presto-hive/src/main/java/com/facebook/presto/hive/HiveDirectoryContext.java

```diff
@@ -13,17 +13,26 @@
  */
 package com.facebook.presto.hive;
 
+import java.util.Map;
+
 import static java.util.Objects.requireNonNull;
 
 public class HiveDirectoryContext
 {
+    public static final String PRESTO_QUERY_ID = "presto_query_id";
+    public static final String PRESTO_QUERY_SOURCE = "presto_query_source";
+    public static final String PRESTO_CLIENT_INFO = "presto_client_info";
+    public static final String PRESTO_USER_NAME = "presto_user_name";
+    public static final String PRESTO_METASTORE_HEADER = "presto_metastore_header";
     private final NestedDirectoryPolicy nestedDirectoryPolicy;
     private final boolean cacheable;
+    private final Map<String, String> additionalProperties;
 
-    public HiveDirectoryContext(NestedDirectoryPolicy nestedDirectoryPolicy, boolean cacheable)
+    public HiveDirectoryContext(NestedDirectoryPolicy nestedDirectoryPolicy, boolean cacheable, Map<String, String> additionalProperties)
```

Comment:
nit: should this be a property bag or an Object. Since this has clearly defined properties and only way to change the property is here by making the code changes, would an Object be better here ?

Reply:
I am a bit skeptical on using Object since object can make the contract too open and sometimes it can be misused to insert mutable entries.